pkgs/build-support/fetchdata/default.nix

```diff
@@ -0,0 +1,36 @@
+{ fetchers }:
+
+# One fetcher to rule them all.
+#
+# The input should only be pure nix data, no functions.
+#
+# Example usage:
+#
+#    fetchurl {
+#      url = "http://example.com/archive.tar.gz";
+#      sha256 = "xxx";
+#    };
+#
+# Becomes:
+#
+#    fetchdata {
+#      type = "fetchurl";
+#      url = "http://example.com/archive.tar.gz";
+#      sha256 = "xxx";
+#    }
+#
+# Which means that the attributes can now be imported
+#
+#    fetchdata (import ./source.nix);
+#
+# That's the whole point of this function.
+#
+{ fetcher ? "fetchurl", ... }@attrs:
```

Comment:
Passing functions as strings seems really ugly to me...

Also, there is nothing in the `fetchdata` function specific to fetching. Rather it's a generic "call a function specified by the 'type' argument". But such a function doesn't sound like a good idea to me. Why write `f { type = "g"; args... }` when I can write `g args` directly?